test/e2e/storage_network_policy.go

```diff
@@ -0,0 +1,307 @@
+				Platform:       "azure",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "azure-file-csi-driver-operator",
+				Platform:       "azure",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "azure-file-csi-driver-controller",
+				Platform:       "azure",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "gcp-pd-csi-driver-operator",
+				Platform:       "gcp",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "gcp-filestore-csi-driver-operator",
+				Platform:       "gcp",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "vmware-vsphere-csi-driver-operator",
+				Platform:       "vsphere",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "vmware-vsphere-csi-driver-controller",
+				Platform:       "vsphere",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "ibm-vpc-block-csi-driver-operator",
+				Platform:       "ibmcloud",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "ibm-vpc-block-csi-controller",
+				Platform:       "ibmcloud",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "openstack-cinder-csi-driver-operator",
+				Platform:       "openstack",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "openstack-cinder-csi-driver-controller",
+				Platform:       "openstack",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "manila-csi-driver-operator",
+				Platform:       "openstack",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      ManilaCSINamespace,
+				Name:           "openstack-manila-csi-controllerplugin",
+				Platform:       "openstack",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "smb-csi-driver-operator",
+				Platform:       "all",
+				RequiredLabels: CSIOperatorRequiredLabels,
+			},
+			{
+				Namespace:      CSINamespace,
+				Name:           "smb-csi-driver-controller",
+				Platform:       "all",
+				RequiredLabels: CSIControllerRequiredLabels,
+			},
+		}
+		runDeploymentChecks(kubeClient, CSIdeploymentsToCheck, currentPlatform)
+	})
+
+	g.It("should verify required pod-selector for networkpolicy in openshift-cluster-storage-operator namespace", func() {
+		npList, err := kubeClient.NetworkingV1().NetworkPolicies(CSONamespace).List(context.TODO(), metav1.ListOptions{})
```

Comment:
The check seems should be firstly check the npList conatins the built-in ones
```console
$ oc get networkpolicy
NAME                                POD-SELECTOR                                              AGE
allow-all-egress                    openshift.storage.network-policy.all-egress=allow         6h27m
allow-egress-to-api-server          openshift.storage.network-policy.api-server=allow         6h27m
allow-ingress-to-operator-metrics   openshift.storage.network-policy.operator-metrics=allow   6h27m
allow-to-dns                        openshift.storage.network-policy.dns=allow                6h27m
default-deny                        <none>                                                    6h27m
```
and then check their `pod-selector` as expected, right?